base/ca/shared/profiles/ca/estServiceCert.cfg

```diff
@@ -124,24 +124,38 @@ policyset.serverCertSet.8.constraint.params.signingAlgsAllowed=SHA256withRSA,SHA
 policyset.serverCertSet.8.default.class_id=signingAlgDefaultImpl
 policyset.serverCertSet.8.default.name=Signing Alg
 policyset.serverCertSet.8.default.params.signingAlg=-
-policyset.serverCertSet.20.constraint.class_id=subjectNameConstraintImpl
+# This policy verify the requested subject match with the client
+# certficiate subject provided to the RA or with client name/uid.
+#
+# If the match has to be peroformed only by EST the following constraint
+# `21` can be used. Replace `20` with `21` in `policyset.serverCertSet.list`.
+policyset.serverCertSet.20.constraint.class_id=raClientAuthSubjectNameConstraintImpl
 policyset.serverCertSet.20.constraint.name=Subject Name Constraint
 # If the subject names in your EST application follow a different pattern,
 # update this parameter to match your application.
-policyset.serverCertSet.20.constraint.params.pattern=CN=[^,]+
-policyset.serverCertSet.20.constraint.params.accept=true
+policyset.serverCertSet.20.constraint.params.pattern=CN=$ra_client_name$
 policyset.serverCertSet.20.default.class_id=subjectNameDefaultImpl
 policyset.serverCertSet.20.default.name=Subject Name Default
 policyset.serverCertSet.20.default.params.name=CN=$request.req_subject_name.cn$
-# This component (`21`) copies the Common Name value to the Subject
+# 
+policyset.serverCertSet.21.constraint.class_id=subjectNameConstraintImpl
+policyset.serverCertSet.21.constraint.name=Subject Name Constraint
+# If the subject names in your EST application follow a different pattern,
+# update this parameter to match your application.
+policyset.serverCertSet.21.constraint.params.pattern=CN=[^,]+
+policyset.serverCertSet.21.constraint.params.accept=true
+policyset.serverCertSet.21.default.class_id=subjectNameDefaultImpl
+policyset.serverCertSet.21.default.name=Subject Name Default
+policyset.serverCertSet.21.default.params.name=CN=$request.req_subject_name.cn$
+# This component (`22`) copies the Common Name value to the Subject
 # Alternative Name extension (as a dNSName value) if and only if it
 # has the appearance of a DNS name (with more than one label/component).
 #
-# To disable this extension remove `21` from the
+# To disable this extension remove `22` from the
 # `policyset.serverCertSet.list` parameter.
-policyset.serverCertSet.21.constraint.class_id=noConstraintImpl
-policyset.serverCertSet.21.constraint.name=No Constraint
-policyset.serverCertSet.21.default.class_id=commonNameToSANDefaultImpl
-policyset.serverCertSet.21.default.name=CN To SAN Default
-policyset.serverCertSet.list=1,2,3,4,6,7,8,10,20,21
+policyset.serverCertSet.22.constraint.class_id=noConstraintImpl
+policyset.serverCertSet.22.constraint.name=No Constraint
+policyset.serverCertSet.22.default.class_id=commonNameToSANDefaultImpl
+policyset.serverCertSet.22.default.name=CN To SAN Default
+policyset.serverCertSet.list=1,2,3,4,6,7,8,10,20,22
```

Comment:
Do we need an upgrade script for these changes?

Reply:
I am not sure here. The current instance should work and have the same behaviour after the upgrade. If they have some customisation in place the upgrade script could create more problems than solution. We can do some tests later for the upgrade to verify if it is needed.

Reply:
I'll let you and @ladycfu decide about this. My concern is that if someone installs EST with PKI 11.6 then later upgrade to PKI 11.7 they won't automatically get the new behavior introduced in PKI 11.7 unless they manually update their config files. Is that OK? Maybe the changes should be documented so people can update their config files manually.

Reply:
Yes, when these changes are completed the documentation will be updated. We can explain what and if should be modified.